pkg/lightning/backend/kv/context.go

```diff
@@ -0,0 +1,260 @@
+	exprCtx               *litExprContext
+	encodingConfig        tbctx.RowEncodingConfig
+	mutateBuffers         *tbctx.MutateBuffers
+	shardID               *variable.RowIDShardGenerator
+	reservedRowIDAlloc    stmtctx.ReservedRowIDAlloc
+	enableMutationChecker bool
+	assertionLevel        variable.AssertionLevel
+	tableDelta            struct {
+		sync.Mutex
+		m map[int64]variable.TableDelta
+	}
+}
+
+// AlternativeAllocators implements the `table.MutateContext` interface.
+func (*litTableMutateContext) AlternativeAllocators(*model.TableInfo) (autoid.Allocators, bool) {
+	// lightning does not support temporary tables, so we don't need to provide alternative allocators.
+	return autoid.Allocators{}, false
+}
+
+// GetExprCtx implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetExprCtx() exprctx.ExprContext {
+	return ctx.exprCtx
+}
+
+// ConnectionID implements the `table.MutateContext` interface.
+func (*litTableMutateContext) ConnectionID() uint64 {
+	// Just return 0 because lightning import does not in any connection.
+	return 0
+}
+
+// InRestrictedSQL implements the `table.MutateContext` interface.
+func (*litTableMutateContext) InRestrictedSQL() bool {
+	// Just return false because lightning import does not in any SQL.
+	return false
+}
+
+// TxnAssertionLevel implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) TxnAssertionLevel() variable.AssertionLevel {
+	return ctx.assertionLevel
+}
+
+// EnableMutationChecker implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) EnableMutationChecker() bool {
+	return ctx.enableMutationChecker
+}
+
+// GetRowEncodingConfig implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetRowEncodingConfig() tbctx.RowEncodingConfig {
+	return ctx.encodingConfig
+}
+
+// GetMutateBuffers implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetMutateBuffers() *tbctx.MutateBuffers {
+	return ctx.mutateBuffers
+}
+
+// GetRowIDShardGenerator implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetRowIDShardGenerator() *variable.RowIDShardGenerator {
+	return ctx.shardID
+}
+
+// GetReservedRowIDAlloc implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetReservedRowIDAlloc() (*stmtctx.ReservedRowIDAlloc, bool) {
+	return &ctx.reservedRowIDAlloc, true
+}
+
+// GetBinlogSupport implements the `table.MutateContext` interface.
+func (*litTableMutateContext) GetBinlogSupport() (tbctx.BinlogSupport, bool) {
+	// lightning import does not support binlog.
+	return nil, false
+}
+
+// GetStatisticsSupport implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetStatisticsSupport() (tbctx.StatisticsSupport, bool) {
+	return ctx, true
+}
+
+// UpdatePhysicalTableDelta implements the `table.StatisticsSupport` interface.
+func (ctx *litTableMutateContext) UpdatePhysicalTableDelta(
```

Comment:
I think lightning don't need implement whole logic. It only need column size by GetColumnSize. Or if we can maintain column size elsewhere this function can be a no-op.

Reply:
I've simplified the implementation, please take a look again.